cpp/src/rolling/rolling.cu

```diff
@@ -217,12 +218,12 @@ void gpu_rolling(column_device_view input,
     //       This might require separating the kernel into a special version
     //       for dynamic and static sizes.
 
-    bool output_is_valid = process_rolling_window<InputType, OutputType, agg_op,
+    volatile bool output_is_valid = false;
```

Comment:
Interesting.  Was the failure case only happening when arg_min_max was false?

Reply:
yes

Reply:
So have you removed support for arg_min/arg_max?

Reply:
I remember @nsakharnykh had a very similar bug in the original rolling implementation that required adding a `volatile` variable.

Reply:
> So have you removed support for arg_min/arg_max?

Yes, I felt it was just an additional over-head. Now I am returning true from `process_rolling_window` for argmin/argmax.

Reply:
> I remember @nsakharnykh had a very similar bug in the original rolling implementation that required adding a `volatile` variable.

I wanted to stop any optimization between call `process_rolling_window` and `__ballot_sync`, and volatile seemed perfect for the job. But I don't know about the bug that @nsakharnykh faced, but it helped as well :).

Reply:
@harrism argmin/argmax support is still there, just the way it's getting handled changed

Reply:
Yes, in my original rolling window code I hit a compiler issue that showed up in CI tests as well. Declaring one of the variables as volatile helped to fix it for CUDA 10.0. I couldn’t reproduce the issue starting from 10.1+ and the compiler team didn’t want to follow up since the issue is no longer present in the latest binaries (we couldn’t really figure what change in the compiler helped to fix it in 10.1). I remember I dumped assembler code for both cases, but it was not clear what’s the cause. If needed I can revive some old emails on this. Let me know.